src/core/font_substitutions.js

```diff
@@ -0,0 +1,454 @@
+  ],
+  [
+    "Calibri-BoldItalic",
+    {
+      local: {
+        alias: "Calibri",
+        append: "Bold Italic",
+      },
+      style: BOLDITALIC,
+      fallback: "Helvetica-BoldItalic",
+    },
+  ],
+]);
+
+/**
+ * Create the src path to use to load a font (see FontFace).
+ * @param {Array<String>} prepend A list of font names to search first.
+ * @param {Array<String>|Object} local A list of font names to search. If an
+ *   Object is passed, then local.alias is the name of an other substition font
+ *   and local.append is a String to append to the list of fonts in the alias.
+ *   For example if local.alias is "Foo" and local.append is "Bold" then the
+ *   list of fonts will be "FooSubst1 Bold", "FooSubst2 Bold", etc.
+ * @returns an String with the local fonts.
+ */
+function makeLocal(prepend, local) {
+  let append = "";
+  if (!Array.isArray(local)) {
+    // We are getting our list of fonts in the alias and we'll append Bold,
+    // Italic or both.
+    append = ` ${local.append}`;
+    local = substitutionMap.get(local.alias).local;
+  }
+  let prependedPaths = "";
+  if (prepend) {
+    prependedPaths = prepend.map(name => `local(${name})`).join(",") + ",";
+  }
+  return (
+    prependedPaths + local.map(name => `local(${name}${append})`).join(",")
+  );
+}
+
+/**
+ * Get a font substitution for a given font.
+ * The general idea is to have enough information to create a CSS rule like
+ * this:
+ *   @font-face {
+ *    font-family: 'Times';
+ *    src: local('Times New Roman'), local('Subst1'), local('Subst2'),
+ *         url(.../TimesNewRoman.ttf)
+ *    font-weight: normal;
+ *    font-style: normal;
+ *   }
+ * or use the FontFace API.
+ *
+ * @param {Object} idFactory The ids factory.
+ * @param {String} localFontPath Path to the fonts directory.
+ * @param {String} baseFontName The font name to be substituted.
+ * @param {String} standardFontName The standard font name to use if the base
+ *   font is not available.
+ * @returns an Object with the CSS, the loaded name, the src and the style.
+ */
+function getFontSubstitution(
+  idFactory,
+  localFontPath,
+  baseFontName,
+  standardFontName
+) {
+  let mustAddBaseFont = false;
+
+  // First, check if we've a substitution for the base font.
+  let substitution = substitutionMap.get(baseFontName);
+  if (!substitution) {
+    // If not, check if we've a substitution for the standard font.
+    substitution = substitutionMap.get(standardFontName);
+    mustAddBaseFont = true;
+  }
+
+  const loadedName = `${idFactory.getDocId()}_sf_${idFactory.createFontId()}`;
+  // If the baseFontName is not valid we don't want to use it.
+  if (!substitution && validateFontName(baseFontName)) {
+    // Maybe we'll be lucky and the OS will have the font.
+    const bold = /bold/gi.test(baseFontName);
+    const italic = /oblique|italic/gi.test(baseFontName);
+    const style =
+      (bold && ((italic && BOLDITALIC) || BOLD)) ||
+      (italic && ITALIC) ||
+      NORMAL;
```

Comment:
While being a tiny bit more code, this would be slightly easier to read at a glance if we format it as follows.
```suggestion
      (bold && italic && BOLDITALIC) ||
      (bold && BOLD)) ||
      (italic && ITALIC) ||
      NORMAL;
```

Reply:
I did that too but I realized that my version has only 2 checks to go to NORMAL when the other one has 3.
But I completely agree with you about the readability so let's go for it.